papers/static/papers/refresh.js

```diff
@@ -172,7 +172,11 @@ function init_paper_module (config) {
   function updateData (data) {
     config.paperSearchResultsNode.html(data.listPapers)
     updateStats(data.stats)
-    config.nbPapersFoundNode.text(data.nb_results)
+    var thousandSeparator = get_format('THOUSAND_SEPARATOR')
+    // Formatting with thousand separator is coming from
+    // https://stackoverflow.com/a/2901298
+    var formattedNbResults = data.nb_results.toString().replace(/\B(?=(\d{3})+(?!\d))/g, thousandSeparator)
+    config.nbPapersFoundNode.text(formattedNbResults)
```

Comment:
This successfully adds separators to the total number of papers found, but I think we need to add this in `updateStats` as well, to update the individual counts for each category. So it would probably make sense to package this as a function.
(And a call to the JS localization system will be needed here, to add the "papers found")